Feature/#117 팀 랭킹 페이지 반응형 레이아웃

llddang opened the pull request:
### 관련 이슈
- 사이드바 관련 이슈 먼저 봐주세요! #110 
- close #117 

### 작업 요약
- 팀 랭킹페이지를 반응형으로 구현했습니다.
- 팀 랭킹페이지의 slider를 swiper 라이브러리를 사용하는 것으로 바꾸었습니다.
- `npm i` 한 번 하시고 실행해주세요.
- 슬라이더 안에 들어가는 팀카드는 새로운 브랜치 파서 작업할 예정입니다!

### 리뷰 요구 사항
- 제 컴퓨터에서 확인을 하긴 했는데요, 여러분의 컴퓨터에서도 제대로 되는 것인지 궁금합니다.
- 리뷰예상 시간 : `10분`

### 미리 보기
<img width="1440" alt="스크린샷 2024-03-10 오후 10 11 24" src="https://github.com/BDD-CLUB/01-doo-re-front/assets/77055208/d36aa963-31a4-4948-a934-72691fdb5b17">

llddang committed:
design: 반응형을 위한 font style 적용

#110

llddang committed:
feat: 사이드바의 CaregoryButton의 isTeam, isTeamMatch props의 기본값을 false로 주기.

#110

llddang committed:
chore: 반응형에 따라서 크기가 변경되도록 icon button size 재설정

#112

llddang committed:
Merge branch 'feature/#112-반응형_버튼_크기_지정' of https://github.com/BDD-CLUB/01-doo-re-front into feature/#110-사이드바_반응형_수정

llddang committed:
refactor: 사이드바의 햄버거 크기 설정을 좀 더 보기 좋게 변경함.

#110

llddang committed:
design: side bar에서 width 속성 삭제

- width이 고정이 아닌, fit-content로 사용됨.
#110

llddang committed:
Merge branch 'develop' of https://github.com/BDD-CLUB/01-doo-re-front into feature/#110-사이드바_반응형_수정

# Conflicts:
#	src/theme/components/button.ts

pipisebastian committed:
design: 사이드바 모바일 반응형 구현

#110

pipisebastian committed:
design: 모바일 사이드바 열려있을때 내부 콘텐츠 밀리도록 수정
#110

llddang committed:
feat: 화면의 크기에 따른 반응형 조절 방식 변경

- 110

llddang committed:
design: 팀 랭킹페이지의 소개글 반응형 구현

#117

llddang committed:
refactor: 팀 랭킹 페이지의 페이지 소개글에서 사용되지 않는 css 삭제

#117

pipisebastian committed:
design: 사이드바 초기상태 닫히도록 수정 및 배경색 추가
#110

llddang committed:
design: 팀 랭킹 페이지에서 페이지 소개글의 border 삭제

- 레이아웃 확인하기 위해서 추가했던 border 속성 삭제
- #117

llddang committed:
chore: react 컴포넌트 - swiper 설치

- #117

llddang committed:
design: 팀 랭킹페이지의 소개글 textStyles 값 변경

- 페이지에서 보기에 글씨가 너무 커보여서 바꾸었습니다.
- #117

llddang committed:
chore: 팀 카드의 props 추가 (idx)

- swiper에서 index로 slide를 넘겨주는데, pagination 과 연결시켜주기 위해 추가했습니다.
- #117

llddang committed:
design: 팀 랭킹페이지의 slide 부분 수정

- 기존에는 제가 작성한 slide 를 넣어는데요, swiper 라이브러리를 알게되어 변경했습니다.
- #117

llddang committed:
refactor: 주석 삭제

- #117

llddang committed:
Merge branch 'feature/#110-사이드바_반응형_수정' of https://github.com/BDD-CLUB/01-doo-re-front into feature/#117-팀_랭킹_페이지_반응형_레이아웃

llddang committed:
fix: 사이드바 zindex 범위 조정

- 사이드 바 전체에 zIndex = 99로 설정하니, 뒤에 있는 요소들을 선택할 수가 없음.
- 그래서 화면의 width이 768px보다 작고 사이드바가 펼쳐졌을 때만, zIndex = 2로 설정.
- #110

llddang committed:
refactor: 사이드바 배경과, 사이드바 통합.

- #110

llddang committed:
Merge branch 'feature/#110-사이드바_반응형_수정' of https://github.com/BDD-CLUB/01-doo-re-front into feature/#117-팀_랭킹_페이지_반응형_레이아웃

llddang committed:
feat: 팀 랭킹 페이지 반응형 구현

- #117

jasper200207 reviewed (approved):
👍

yeonddori reviewed (approved):
제 노트북에서도 잘 돌아가네요👍

pipisebastian commented:
이게 develop사진이고
![image](https://github.com/BDD-CLUB/01-doo-re-front/assets/81643702/dc8e16f6-fa56-46b8-b629-665c2e4225fb)

이게 땅님 브랜치에서 찍은건데,
![image](https://github.com/BDD-CLUB/01-doo-re-front/assets/81643702/419104be-c621-49b7-8b6d-692cc0d1a2c6)
밑에 보시면 스크롤바?인지, 흰색 바가 존재합니다,,! 혹시 이거 없애주실 수 있나용? 
ㅇㅁㅇ.. 갑자기 스크롤바가 안보이는걸요....

mun-kyeong commented:
<img width="959" alt="image" src="https://github.com/BDD-CLUB/01-doo-re-front/assets/96461334/499a88ac-ad8c-4216-8176-e79dd982fc6f">
@pipisebastian  스크롤바 맞는 것 같습니다! 제 화면에선 이렇게 뜨네욤

pipisebastian commented:
![image](https://github.com/BDD-CLUB/01-doo-re-front/assets/81643702/614b87c8-7f4e-4379-b8fa-9e1d6fd55f68)
이게 랭킹페이지 100%인 상태에서 80%로 줄였을때, 화면을 꽉채워서 줄어지는게 아니라, 위에까지 같이 보입니당....🥺

llddang committed:
design: 가로 스크롤바 없애기

- overflow: scroll 로 설정하면 안의 내용에 상관없이 가로, 세로 스크롤바가 무조건 생성됩니다.
- overflow: auto 로 설정하면 안의 내용 크기에 따라 스크롤바가 생깁니다.
- #117

llddang committed:
design: swiper가 부모요소의 크기 넘지 않도록 설정

- #117

llddang commented:
> 이게 develop사진이고 ![image](https://private-user-images.githubusercontent.com/81643702/312509137-dc8e16f6-fa56-46b8-b629-665c2e4225fb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA0MDk5OTEsIm5iZiI6MTcxMDQwOTY5MSwicGF0aCI6Ii84MTY0MzcwMi8zMTI1MDkxMzctZGM4ZTE2ZjYtZmE1Ni00NmI4LWI2MjktNjY1YzJlNDIyNWZiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzE0VDA5NDgxMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY4YmIxY2I3OGY0MDg4ZWM3ZTU2ZjkxNmNmMDE5ZDRjNjhiYWY3ZTYwZjRhNzg1YWJlYjc5MWE1YmVkYzU2MjcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.7c-iDed_Jk19WsK6UQJmIsdjtywLGc5PlSNAenf1HVw)
> 
> 이게 땅님 브랜치에서 찍은건데, ![image](https://private-user-images.githubusercontent.com/81643702/312510085-419104be-c621-49b7-8b6d-692cc0d1a2c6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA0MDk5OTEsIm5iZiI6MTcxMDQwOTY5MSwicGF0aCI6Ii84MTY0MzcwMi8zMTI1MTAwODUtNDE5MTA0YmUtYzYyMS00OWI3LThiNmQtNjkyY2MwZDFhMmM2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzE0VDA5NDgxMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJjOTc1NTI4OTJiZmU4ZTU1MjMwZDYzNzJkMzEzZjQ3MzMzNDkzM2NkYzQxMWJhNTE1ZTFlMzY2NjNiNWYxMzMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.qb2LYM-eIV_ARqyltuWXLs1mMg1XIE6EL6rnJrPtYck) 밑에 보시면 스크롤바?인지, 흰색 바가 존재합니다,,! 혹시 이거 없애주실 수 있나용? ㅇㅁㅇ.. 갑자기 스크롤바가 안보이는걸요....

`MainSection`에서 `overflow: scroll`로 설정하여 가로 세로 스크롤이 보였던 것이였습니다.
`overflow: auto`로 설정하여 안의 내용의 크기에 따라 스크롤이 생기도록 조정했습니다.

llddang committed:
fix: 화면의 세로 길이를 늘렸을 때, 윗부분 세션이 보이는 에러 수정

- #117

llddang commented:
> ![image](https://private-user-images.githubusercontent.com/81643702/312711304-614b87c8-7f4e-4379-b8fa-9e1d6fd55f68.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA0MDk5OTEsIm5iZiI6MTcxMDQwOTY5MSwicGF0aCI6Ii84MTY0MzcwMi8zMTI3MTEzMDQtNjE0Yjg3YzgtN2Y0ZS00Mzc5LWI4ZmEtOWUxZDZmZDU1ZjY4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzE0VDA5NDgxMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTUzOWE5YzIxZWZhMTMxMmMwYjVlNGQ4MWE5NzdlZGNkMDc3NmExMjA2MDEwZDMyZTc1MDIxYmNhZjFhMWZlNWImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.bRao0Wsaf2TANdtcQnXW11hiNrWqq7Yqs-WklA1PTxE) 이게 랭킹페이지 100%인 상태에서 80%로 줄였을때, 화면을 꽉채워서 줄어지는게 아니라, 위에까지 같이 보입니당....🥺

화면이 비율이 변할 때마다, section의 index를 현재 section의 Index로 설정하여 해결했습니다!

mun-kyeong commented:
제 화면사이즈 1280px X 585.33px 입니답!
화면이 작아서 화면 가로스크롤이 나왔던게 아닌가 하는 생각이 드네요

mun-kyeong reviewed (commented):
제 화면사이즈 1280px X 585.33px 입니답!
화면이 작아서 화면 가로스크롤이 나왔던게 아닌가 하는 생각이 드네요

pipisebastian reviewed (approved):
굿..

llddang committed:
Merge branch 'develop' of https://github.com/BDD-CLUB/01-doo-re-front into feature/#117-팀_랭킹_페이지_반응형_레이아웃

# Conflicts:
#	src/app/layout.tsx
